File: VideoService-Play-Java-Ebean-NoAuth/app/videosvc/Global.java
package videosvc;

import play.Application;
import play.GlobalSettings;
import play.Logger;
import play.libs.F;
import play.mvc.Action;
import play.mvc.Http;
import play.mvc.Result;

import java.lang.reflect.Method;
import java.lang.reflect.Parameter;
import java.util.Map;
import java.util.Set;

public class Global extends GlobalSettings {

    private final Logger.ALogger l = Logger.of(getClass());


    @Override
    public void beforeStart(Application app) {
        l.debug("===> WebService starting up ...");
    }

    @Override
    public void onStart(Application app) {
        l.debug("===> WebService has started");
    }

    @Override
    public void onStop(Application app) {
        l.debug("<=== WebService shutting down ...\n");
    }

    @Override
    public F.Promise<Result> onError(Http.RequestHeader request, Throwable t) {
        l.error("! Exception while processing request: " + request);
        l.error(t.getClass().getSimpleName() + ": message = " + t.getMessage(), t);
        return super.onError(request, t);
    }

    @Override
    public Action onRequest(Http.Request request, Method actionMethod) {

        String paramList = parameterList(actionMethod);
        l.debug("---> Request (" + request.method() + " " + request.path() + ") => "
                + actionMethod.getName() + "(" + paramList + ")");

        showRequestForAddVideo(request, actionMethod);

        return super.onRequest(request, actionMethod);
    }

    private void showRequestForAddVideo(Http.Request request, Method actionMethod) {
        if (actionMethod.getName().equals("addVideo")) {        // --nonExistingMethod
            showRequest(request);
        }
    }

    private void showRequest(Http.Request request) {
        l.debug("------------------------------------------------------------------");
        Set<Map.Entry<String, String[]>> headers = request.headers().entrySet();
        for (Map.Entry<String, String[]> header : headers) {
            String key = header.getKey();
            String[] values = header.getValue();
            for (String value : values) {
                l.debug("HEADER - " + key + ": " + value);
            }
        }
//        request.headers().entrySet().forEach(hs -> {
//            for (String value : hs.getValue()) {
//                l.debug("HEADER - " + hs.getKey() + ": " + value);
//            }
//        });
//        request.headers().entrySet().forEach(hs -> {
//            Arrays.stream(hs.getValue()).forEach(v -> l.debug("HEADER - " + hs.getKey() + ": " + v));
//        });
        l.debug("BODY: " + request.body().toString());
        l.debug("------------------------------------------------------------------");
    }

    private String parameterList(Method actionMethod) {

        int count = actionMethod.getParameterCount();
        Class<?>[] types = actionMethod.getParameterTypes();
        Parameter[] params = actionMethod.getParameters();

        String paramList = "";
        for ( int i = 0; i < count; i++) {
            paramList += params[i].getType();   // + " " + params[i].getName();
            if (i < params.length - 1) {
                paramList += ", ";
            }
        }

        return paramList;
    }
}

File: VideoService-Play-Scala-Slick-NoAuth/app/videosvc/Global.scala
package videosvc

import play.api.db.slick.DatabaseConfigProvider
import play.api.{Logger, Play, Application, GlobalSettings}
import play.api.mvc.{Handler, Result, RequestHeader}
import videosvc.models._

import scala.concurrent.duration.Duration
import scala.concurrent.{Await, Future}
import slick.driver.JdbcProfile

case object Global extends GlobalSettings {

  import slick.driver.H2Driver.api._

  val l: Logger = Logger(this.getClass())

  override def beforeStart(app: Application): Unit = {
    l.debug("===> WebService starting up ...")
  }

  override def onStart(app: Application): Unit = {
    l.debug("===> WebService has started")
    l.debug("===> Creating Database Schema ...")
    val db = DatabaseConfigProvider.get[JdbcProfile](Play.current).db
    Await.result(db.run(
      (TableQuery[Videos].schema ++ TableQuery[UserVideoRatings].schema).create
    ), Duration.Inf)
    l.debug("===> Database Schema created")
  }

  override def onStop(app: Application): Unit = {
    l.debug("===> Dropping Database Schema ...")
    val db = DatabaseConfigProvider.get[JdbcProfile](Play.current).db
    Await.result(db.run(
      (TableQuery[Videos].schema ++ TableQuery[UserVideoRatings].schema).drop
    ), Duration.Inf)
    l.debug("<=== Database Schema dropped")
    l.debug("<=== WebService shutting down ...\n")
  }

  override def onRequestReceived(request: RequestHeader): (RequestHeader, Handler) = {
    super.onRequestReceived(request)
  }

  override def onError(request: RequestHeader, ex: Throwable): Future[Result] = {
    super.onError(request, ex)
  }
}

